docs/book/src/developer/providers/contracts/infra-machinepool.md

```diff
@@ -0,0 +1,544 @@
+
+For further information see the [proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220209-machinepool-machines.md).
+
+### InfraMachinePool: providerID
+
+Each InfraMachinePool MAY specify a provider ID on `spec.providerID` that can be used to identify the infrastructure resource that implements the InfraMachinePool.
+
+This field isn't used by core CAPI. Its main purpose is purely informational to the user to surface the infrastructures identifier for the InfraMachinePool. For example, for AWSMachinePool this would be the ASG identifier.
+
+```go
+type FooMachinePoolSpec struct {
+    // providerID is the identification ID of the FooMachinePool.
+    // +optional
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=512
+    ProviderID string `json:"providerID,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+NOTE: To align with API conventions, we recommend since the v1beta2 contract that the `ProviderID` field should be
+of type `string`.
+
+### InfraMachinePool: providerIDList
+
+Each InfraMachinePool MUST supply a list of the identification IDs of the machine instances managed by the machine pool by storing these in `spec.providerIDList`.
+
+```go
+type FooMachinePoolSpec struct {
+    // ProviderIDList is the list of identification IDs of machine instances managed by this Machine Pool
+    // +optional
+    // +listType=atomic
+    // +kubebuilder:validation:MaxItems=10000
+    // +kubebuilder:validation:items:MinLength=1
+    // +kubebuilder:validation:items:MaxLength=512
+    ProviderIDList []string `json:"providerIDList,omitempty"`
+    
+    // See other rules for more details about mandatory/optional fields in InfraMachinePool spec.
+    // Other fields SHOULD be added based on the needs of your provider.
+}
+```
+
+Cluster API uses this list to determine the status of the machine pool and to know when replicas have been deleted, at which point the Node will be deleted. Therefore, the list MUST be kept up to date.
+
+### InfraMachinePool: initialization completed
+
+Each provider MUST indicate when then the InfraMachinePool has been completely provisioned.
+
+Currently this is done by setting `status.ready` to **true**.  The value returned here is stored in the MachinePool's `status.infraStructureReady` field.
+
+Additionally providers should set `initialization.provisioned` to **true**. This value isn't currently used by core CAPI for MachinePools. However, MachinePools will start to use this instead and `status.ready` will be deprecated. By setting both these fields it will make the future migration easier.
```

Comment:
I'm not sure asking providers to have both provides benefit, e.g. for other contract we ask to have one, depending on the contract they declare the provider is supporting.
 
Also, do you have a timeline in mind for for this to happen?

(for sake of simplicity I would prefer if there is only one way to transition from one contract version to another; similarly it will be easier to manage if we can align the timeline to v1beta1 removal)